homeassistant/components/alert.py

```diff
@@ -0,0 +1,272 @@
+"""
+Support for repeating alerts when conditions are met.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/alert/
+"""
+import asyncio
+from datetime import datetime, timedelta
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (CONF_ENTITY_ID, STATE_IDLE, CONF_NAME,
+                                 CONF_STATE, STATE_ON, STATE_OFF,
+                                 SERVICE_TURN_ON, SERVICE_TURN_OFF,
+                                 SERVICE_TOGGLE, ATTR_ENTITY_ID)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers import service, event
+from homeassistant.util import slugify
+from homeassistant.util.async import run_callback_threadsafe
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'alert'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+CONF_CAN_ACK = 'can_acknowledge'
+CONF_NOTIFIERS = 'notifiers'
+CONF_REPEAT = 'repeat'
+CONF_SKIP_FIRST = 'skip_first'
+
+ALERT_SCHEMA = vol.Schema({
+    # pylint: disable=no-value-for-parameter
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_ENTITY_ID): cv.string,
+    vol.Required(CONF_STATE, default=STATE_ON): cv.string,
+    vol.Required(CONF_REPEAT): vol.All(cv.ensure_list, [vol.Coerce(float)]),
+    vol.Required(CONF_CAN_ACK, default=True): cv.boolean,
+    vol.Required(CONF_SKIP_FIRST, default=False): cv.boolean,
+    vol.Required(CONF_NOTIFIERS): cv.ensure_list})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.string: ALERT_SCHEMA,
```

Comment:
Instead of `cv.string` you can require it to be a slug with `cv.slug`. This also makes it compatible with packages. Look at `input_*`, `group`, `automation` etc...